aidial_adapter_bedrock/llm/model/claude/v3/converters.py

```diff
@@ -128,9 +150,50 @@ async def to_claude_messages(
 
     claude_messages: List[MessageParam] = []
     for message in messages:
-        role, message = _to_claude_role(message)
-        content = await _to_claude_content(message, file_storage)
-        claude_messages.append(MessageParam(role=role, content=content))
+        match message:
+            case HumanRegularMessage():
+                claude_messages.append(
+                    MessageParam(
+                        role="user",
+                        content=await _basic_message_to_claude_content(
+                            message, file_storage
+                        ),
+                    )
+                )
+            case AIRegularMessage():
+                claude_messages.append(
+                    MessageParam(
+                        role="assistant",
+                        content=await _basic_message_to_claude_content(
+                            message, file_storage
+                        ),
+                    )
+                )
+            case AIToolCallMessage():
+                claude_messages.append(
+                    MessageParam(
+                        role="assistant",
+                        content=[
+                            _tool_call_to_claude_content(call)
+                            for call in message.calls
+                        ],
+                    )
+                )
+            case HumanToolResultMessage():
+                claude_messages.append(
+                    MessageParam(
+                        role="user",
+                        content=[_tool_result_to_claude_content(message)],
+                    )
+                )
+            case SystemMessage():
+                raise ValueError(
+                    "System message is only allowed as the first message"
+                )
+            case _:
+                raise ValidationError(
+                    f"Unknown type of of message! {type(message)}"
```

Comment:
They are actually known: `HumanFunctionResultMessage | AIFunctionCallMessage`.
The right word is _not supported_. 

Previous legacy tool use did support functions.  So this version should too.

Reply:
Try run integration tests. There are a few tests for tools and functions. They must pass.

Reply:
https://github.com/search?q=repo%3Aepam%2Fai-dial-chat+path%3A%2F%5Eapps%5C%2Fchat-e2e%5C%2Fsrc%5C%2F%2F+tools&type=code


I don't see any tests for tools or functions in our e2e tests

Reply:
Though, you're right, I overlooked support of functions


Added emulation of function requests using native Claude Tools

Fixed